### ISSUE TITLE
Brainstorm 508 scanning functionality

### ISSUE THREAD

**Issue by gbinal** [18F/pulse#259]
We're thinking about what 508/accessibility testing would be useful.  What analysis do we want to run? 

At this stage, this would be analyzing home pages.  cc @adelevie, @nickbristow

**Comment by normanbrobinson** [comment]
Sorry for not providing a short definitive answer, but the basics are in tools such as pa11y (http://pa11y.org/) or HTML_Codesniffer (http://squizlabs.github.io/HTML_CodeSniffer/) or RAAKT (http://peterkrantz.com/raakt/wiki/) however it should be noted that *Section 508* compliance is more explicit than some of the usability items some tools flag. "Throwing it all in there" when claiming to meet 508 can backfire on you, in that one must be clear it is meeting the Access Board's IT Accessibility Standards (required by Section 508) versus more usable or nice to have features. So somewhere between targeting WCAG Level AA (the 508 refresh is expected to state they are equivalent to 508 technical standards) and a validated test process that has traceability to 508 standards (http://www.dhs.gov/compliance-test-processes has this) whatever tool should have a ruleset or configuration based on the harmonized test cases shown in http://www.dhs.gov/publication/dhs-section-508-compliance-test-processes.

**Comment by adelevie** [comment]
Thanks, @normanbrobinson. We're close to getting a `pa11y` scanner built into [`domain-scan`](https://github.com/18F/domain-scan/pull/39) (which powers much of Pulse). We're also working on documenting the [reporting process](https://github.com/18F/pulse/blob/document_a11y_approach/documentation/accessibility.md) for the data returned by `pa11y`. If you have any feedback or ideas about either, we'd love to hear it.

**Comment by normanbrobinson** [comment]
The reporting and traceability to standards, along with ensuring the test is consistently valid is the hardest part!  So in the previously mentioned test processes, the teams not only considered Section 508, but validated each use case against assistive technology use cases to provide techniques and test cases. Reading the baseline (harmonized) test document http://www.dhs.gov/sites/default/files/publications/Baseline_Tests_for_Software_and_Web_Accessibility_0.pdf and looking at how the team collects and reports gives a recommended approach for validating each standard. They map to the WCAG AA as well, so the tracing between testing techniques and WCAG is already done: that is explained and mapped in the testing process appendix. Unfortunately the test sheet is in Excel, but I can probably map to pa11y or assist in validating a workable process against the same content used to validate a repeatable, consistent test.

**Comment by gbinal** [comment]
Closing this as the results are now factored into https://staging.pulse.cio.gov/accessibility/domains/
